### PR TITLE
added how to guide for multi output

### LINKED ISSUE #68: How-to guide to cover multi-output models and their evaluation
The idea is to show how to tackle a use-case like:
```
x, y1, y2 = batch
y_pred1, y_pred2, aux = model(x)
```
and where we would like to compute metrics between `y_pred1` vs `y1` and `y_pred2` vs `y2`.

- Show that we can create custom evaluator that returns a dict
- We attach metric and specify `output_transform` for each metric

Context: discussed on discord, "questions" channel, multi-output thread.

### PR DESCRIPTION
**Pull request by Priyansi**
Fixes #68

### CODE REVIEW

File: how-to-guides/12-train-multi-output-model.ipynb [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+      "source": [
+        "def train_step(engine, batch):\n",
+        "    model.train()\n",
+        "    optimizer.zero_grad()\n",
+        "    x, y1, y2 = batch[0].to(device), batch[1].to(device), batch[2].to(device)\n",
+        "    y_pred1, y_pred2 = model(x)\n",
+        "    loss1 = criterion(y_pred1, y1)\n",
+        "    loss2 = criterion(y_pred2, y2)\n",
+        "\n",
+        "    loss1.backward()\n",
```
Comment (sdesrozis) [inline how-to-guides/12-train-multi-output-model.ipynb:218]: We talked about the tracking of each loss' components to figure out that `x^2` can't be approximate by a `Linear` but `2*x` can. 
```python
loss = loss1 + loss2
loss.backward()

return {"loss": loss.item(), "loss1": loss1.item(), "loss2": loss2.item()}
```

File: how-to-guides/12-train-multi-output-model.ipynb [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+        "        super(Net, self).__init__()\n",
+        "        \n",
+        "        self.model1 = nn.Linear(1, 1)\n",
+        "        self.model2 = nn.Linear(1, 1)\n",
+        "\n",
+        "    def forward(self, x):\n",
+        "        return self.model1(x), self.model2(x)\n",
+        "\n",
+        "model = Net().to(device)\n",
+        "optimizer = torch.optim.Adam(model.parameters(), lr=0.005)\n",
```
Comment (sdesrozis) [inline how-to-guides/12-train-multi-output-model.ipynb:185]:
```suggestion
        "optimizer = torch.optim.Adam(model.parameters(), lr=0.1)\n",
```

File: how-to-guides/12-train-multi-output-model.ipynb [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+      "cell_type": "code",
+      "metadata": {
+        "id": "7_Y_U0cCEp4W"
+      },
+      "source": [
+        "@trainer.on(Events.EPOCH_COMPLETED)\n",
+        "def log_validation_results(trainer):\n",
+        "    val_evaluator.run(val_loader)\n",
+        "    metrics = val_evaluator.state.metrics\n",
+        "    print(f\"Validation Results - Epoch[{trainer.state.epoch}] Avg MAE: {metrics['mae']:.2f}\")"
```
Comment (sdesrozis) [inline how-to-guides/12-train-multi-output-model.ipynb:349]: Add this to track losses
```python
RunningAverage(output_transform=lambda x: x["loss"]).attach(trainer, "loss")
RunningAverage(output_transform=lambda x: x["loss1"]).attach(trainer, "loss1")
RunningAverage(output_transform=lambda x: x["loss2"]).attach(trainer, "loss2")

@trainer.on(Events.EPOCH_COMPLETED)
def iteration_results(trainer):
    metrics = trainer.state.metrics
    print(f"Iteration Results - Epoch[{trainer.state.iteration}] Metrics: {metrics}")
```

**Comment by sdesrozis** [comment]
@Priyansi I left a few comments to track the losses. What do you think ?
